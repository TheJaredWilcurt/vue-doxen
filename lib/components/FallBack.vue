<template>
  <h1 id="fallback">
    OH NO!
    You are missing a needed component!
    View console for details.
  </h1>
</template>

<script>
export default {
  name: 'FallBack',
  created: function () {
    const message = [
      'This fallback component is only used when a custom component',
      'is not supplied in the VueDoxenCustom component\'s options',
      'prop. The following props/attributes were passed in to this',
      'fallback component:'
    ].join(' ');
    const data = {
      ...this.$props,
      ...this.$attrs
    };
    console.info(message, data);
  }
};
</script>

<style scoped>
#fallback {
  position: relative;
  display: block;
  background: #000;
  border: 0px;
  margin: 10px;
  padding: 10px;
  color: #FF0;
  font-size: 20px;
  opacity: 1;
  visibility: visible;
}
</style>
