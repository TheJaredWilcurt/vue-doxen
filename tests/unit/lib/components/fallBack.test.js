import FallBack from '@/components/FallBack.vue';

import testHelpers from '@@/unit/testHelpers.js';

describe('FallBack.vue', () => {
  let consoleInfo;

  beforeEach(() => {
    consoleInfo = console.info;
    console.info = vi.fn();
  });

  afterEach(() => {
    console.info = consoleInfo;
  });

  const setupWrapper = async () => {
    const requiredProps = {};
    const options = {
      props: requiredProps
    };
    const wrapper = await testHelpers.mount(FallBack, options);
    return wrapper;
  };

  test('Renders correctly', async () => {
    const wrapper = await setupWrapper();

    expect(console.info)
      .toHaveBeenCalledWith(
        'This fallback component is only used when a custom component ' +
        'is not supplied in the VueDoxenCustom component\'s options ' +
        'prop. The following props/attributes were passed in to this ' +
        'fallback component:',
        {}
      );

    expect(wrapper)
      .toMatchSnapshot();
  });
});
