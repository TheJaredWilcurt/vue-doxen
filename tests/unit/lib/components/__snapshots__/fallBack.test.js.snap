// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FallBack.vue > Renders correctly 1`] = `<h1 id="fallback"> OH NO! You are missing a needed component! View console for details. </h1>`;
